app/controllers/api/users_controller.rb

```diff
@@ -1,15 +1,24 @@
-class Api::UsersController < Api::BaseController
-  before_action :doorkeeper_authorize!, only: %i[index show]
-
-  def index
-    # inside service params are checked and whiteisted
-    @users = UserService::Index.new(params.permit!, current_resource_owner).execute
-    @total_pages = @users.total_pages
+class Api::UsersController < ApplicationController
+  # Other methods...
+  def register
+    begin
+      user_params = params.require(:user).permit(:username, :email, :password)
+      validate_user_params(user_params)
+      user = UserRegistrationService.new(user_params).register
+      render json: UserSerializer.new(user).serialized_json, status: :ok
```

Comment:
The `Api::UsersController` is using a manual way to require parameters and validate them. Consider using ActiveModel Serializers or similar mechanisms for parameter validation to make the code cleaner and more maintainable.